OrbitQt/main.cpp

```diff
@@ -122,32 +123,44 @@ static outcome::result<void> RunUiInstance(
     service_deploy_manager_ptr = &service_deploy_manager.value();
   }
 
-  OrbitMainWindow w(app, std::move(options), service_deploy_manager_ptr);
-  // "resize" is required to make "showMaximized" work properly.
-  w.resize(1280, 720);
-  w.showMaximized();
-
   std::optional<std::error_code> error;
-  auto error_handler = [&]() -> ScopedConnection {
-    if (!service_deploy_manager) {
-      return ScopedConnection();
+
+  OrbitApp::Init(std::move(options), CreateMainThreadExecutor());
+
+  {  // Scoping of QT UI Resources
+    OrbitMainWindow w(app, service_deploy_manager_ptr);
+
+    Orbit_ImGui_Init();
+
+    // "resize" is required to make "showMaximized" work properly.
+    w.resize(1280, 720);
+    w.showMaximized();
+
+    auto error_handler = [&]() -> ScopedConnection {
+      if (!service_deploy_manager) {
+        return ScopedConnection();
+      }
+
+      return OrbitSshQt::ScopedConnection{QObject::connect(
+          &service_deploy_manager.value(), &ServiceDeployManager::socketErrorOccurred,
+          &service_deploy_manager.value(), [&](std::error_code e) {
+            error = e;
+            w.close();
+            QApplication::quit();
+          })};
+    }();
+
+    if (!capture_path.isEmpty()) {
+      w.OpenCapture(capture_path.toStdString());
     }
 
-    return OrbitSshQt::ScopedConnection{QObject::connect(
-        &service_deploy_manager.value(), &ServiceDeployManager::socketErrorOccurred,
-        &service_deploy_manager.value(), [&](std::error_code e) {
-          error = e;
-          w.close();
-          QApplication::quit();
-        })};
-  }();
-
-  if (!capture_path.isEmpty()) {
-    w.OpenCapture(capture_path.toStdString());
+    QApplication::exec();
+
+    Orbit_ImGui_Shutdown();
```

Comment:
I wonder if you want to have a scoped object (ScopedOrbitImGui) where this is called in the d-tor. There are no early returns atm but someone could add one in the future.

Reply:
Yes, that would be a nice improvement. But I'd leave that for the future, I wanted to keep the changes small since it's a release hotfix. I'll track that in a separate bug.